storage/jellyfish-merkle/src/node_type/mod.rs

```diff
@@ -351,13 +353,9 @@ impl InternalNode {
     /// Given a range [start, start + width), returns the sub-bitmap of that range.
     fn range_bitmaps(start: u8, width: u8, bitmaps: (u16, u16)) -> (u16, u16) {
         assert!(start < 16 && width.count_ones() == 1 && start % width == 0);
+        assert!(width <= 16 && (start + width) <= 16);
         // A range with `start == 8` and `width == 4` will generate a mask 0b0000111100000000.
-        let mask = if width == 16 {
-            0xffff
-        } else {
```

Comment:
How is this being preserved? Is there something implicit in how rust converts u32 to u16?

Reply:
the original code is not easy to understand